extendr-api/src/metadata.rs

```diff
@@ -43,6 +43,37 @@ pub struct Metadata {
     pub impls: Vec<Impl>,
 }
 
+struct RArg {
+    name: String,
+    default: Option<&'static str>,
+}
+
+impl RArg {
```

Comment:
I think it's a bit misleading to call this `RArg` when it's actually representing a *Rust* argument that will be converted into R.

Reply:
I originally tried to expand the functionality of `Arg`, but then realized it is used in many places. Here I want to introduce a struct to contain metadata of R arguments, mainly sanitized name and default value (if present). Hence the name. So it is not really *Rust* argument but *R* argument. 

Maybe we can improve on naming by utilizing the difference between *parameter* and *argument*. Here R parameters are names with optional default values that go into R function definition, and R arguments are actual arguments supplied to the `.Call()` instruction within that wrapper.

I am open to suggestions.

Reply:
Okay I take your point, I think this is a reasonable name.